libdispatch/ncs3sdk_h5.c

```diff
@@ -184,8 +186,10 @@ NC_s3sdkcreateclient(NCS3INFO* info)
 
     s3client = (NCS3CLIENT*)calloc(1,sizeof(NCS3CLIENT));
     if(s3client == NULL) goto done;
-    // We load credentials from env if not in profile
-    NC_s3getcredentials(info->profile, NULL, &accessid, &accesskey);
+    if(info->profile != NULL) {
+        if((stat = NC_s3profilelookup(info->profile, "aws_access_key_id", &accessid))) goto done;
+        if((stat = NC_s3profilelookup(info->profile, "aws_secret_access_key", &accesskey))) goto done;
+    }
```

Comment:
This will cause issues when **not** using a profile and relying on environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
`NC_s3getcredentials` will fallback to the global config (same as the environement) for the case where `profile="no"`

Reply:
@DennisHeimbigner  we should definitely add a test for this case.

Reply:
Indeed! The utility function `find_profile` seems similar to `NC_s3getcredentials`. As that PR is still draft, I guess integration will come later.

For this PR more important would be to get this tested (loading credentials from environment), preferably using the S3 NCAR bucket. As it is, it passes the tests but in reality it breaks behaviour:
```
$ export AWS_ACCESS_KEY_ID=.......; export AWS_SECRET_ACCESS_KEY=......; ncdump -h https://s3.eu-dkrz-1.dkrz.cloud/datenanalyse/k202186/u_4km.nc#mode=s3,bytes
ncdump: https://s3.eu-dkrz-1.dkrz.cloud/datenanalyse/k202186/u_4km.nc#mode=s3,bytes: NetCDF: Authorization failure
```

Reply:
re:find_profile versus NC_s3getcredentials
I will see if they can be unified.
If you see other cases of code duplicatio, let me know.